### ISSUE TITLE
Unexpected tainted behaviour with println

### ISSUE THREAD

**Issue by anupama-pathirage** [ballerina-platform/ballerina-lang#11386]
**Description:**
Following code gives the below error only if the println function is commented out.
```
tainted value passed to sensitive parameter 'sqlQuery'
```
```ballerina
import ballerina/mysql;
import ballerina/io;

endpoint mysql:Client mysqlDB {
    host: "localhost",
    port: 3306,
    name: "ballerinademo",
    username: "demouser",
    password: "password@123",
    dbOptions: { useSSL: false}
};

public function main(string... args) {
    string s1 = args[0];
    io:println(s1);
    //Iterating table
    table t1 = check mysqlDB->select("SELECT id, age, name from employee 
                    where name = " + s1, ());
}
```

**Steps to reproduce:**

**Affected Versions:**
0.983.0
**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by Manuri** [comment]
When debugging the TaintAnalyzer following was observed.

When the println wasn't present, the `taintRecord` was null for the main function argument so error was printed right-away
When it was present, `taintRecord` wasn't null and it went on to the else block but `taintRecord.taintError` was null.

@ayomawdb how the flow is supposed to be?

```java
           if (taintRecord == null) {
                // This is when current parameter is "sensitive". Therefore, providing a tainted value to a sensitive
                // parameter is invalid and should return a compiler error.
                DiagnosticPos argPos = argExpr.pos != null ? argExpr.pos : invocationExpr.pos;
                addTaintError(argPos, paramSymbol.name.value,
                        DiagnosticCode.TAINTED_VALUE_PASSED_TO_SENSITIVE_PARAMETER);
            } else if (taintRecord.taintError != null && taintRecord.taintError.size() > 0) {
                // This is when current parameter is derived to be sensitive.
                taintRecord.taintError.forEach(error -> {
                    if (error.diagnosticCode == DiagnosticCode.TAINTED_VALUE_PASSED_TO_GLOBAL_VARIABLE) {
                        addTaintError(taintRecord.taintError);
                    } else {
                        DiagnosticPos argPos = argExpr.pos != null ? argExpr.pos : invocationExpr.pos;
                        addTaintError(argPos, paramSymbol.name.value,
                                DiagnosticCode.TAINTED_VALUE_PASSED_TO_SENSITIVE_PARAMETER);
                    }
                });
```

**Comment by ayomawdb** [comment]
PR #11395 resolves this issue. Please check the PR for further details. 

Clarified the issue to @Manuri offline.

**Comment by ayomawdb** [comment]
This is fixed with #11584. @anupama-pathirage shall we close the issue?
